File: frontEnd/Systetica_App/lib/model/Empresa.dart
// ignore_for_file: file_names

import 'Cidade.dart';
import 'Usuario.dart';

class Empresa {
  Empresa({
    this.id,
    this.nome,
    this.cnpj,
    this.telefone1,
    this.telefone2,
    this.endereco,
    this.numero,
    this.cep,
    this.bairro,
    this.latitude,
    this.longitude,
    this.logoBase64,
    this.cidade,
    this.usuarioAdministrador,
    this.nomeUsuario,
    this.horarioAbertura,
    this.horarioFechamento,
    this.usuariosFuncionario,
  });

  int? id;
  String? nome;
  String? cnpj;
  String? telefone1;
  String? telefone2;
  String? endereco;
  int? numero;
  String? cep;
  String? bairro;
  String? latitude;
  String? longitude;
  String? logoBase64;
  Cidade? cidade;
  Usuario? usuarioAdministrador;
  String? nomeUsuario;
  String? horarioAbertura;
  String? horarioFechamento;
  List<Usuario>? usuariosFuncionario;

  Empresa.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    cnpj = json['cnpj'];
    telefone1 = json['telefone1'];
    telefone2 = json['telefone2'];
    endereco = json['endereco'];
    numero = json['numero'];
    cep = json['cep'];
    bairro = json['bairro'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    logoBase64 = json['logo_base64'];
    cidade = Cidade.fromJson(json['cidade']);
    nomeUsuario = json['nome_usuario'];
    horarioAbertura = json['horario_abertura'];
    horarioFechamento = json['horario_fechamento'];
    usuariosFuncionario = Usuario.fromJsonList(json['usuarios']);
  }

  static List<Empresa> fromJsonList(List json) {
    return json.map((item) => Empresa.fromJson(item)).toList();
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    _data['nome'] = nome;
    _data['cnpj'] = cnpj;
    _data['telefone1'] = telefone1;
    _data['telefone2'] = telefone2;
    _data['endereco'] = endereco;
    _data['numero'] = numero;
    _data['cep'] = cep;
    _data['bairro'] = bairro;
    _data['latitude'] = latitude;
    _data['longitude'] = longitude;
    _data['logo_base64'] = logoBase64;
    _data['cidade'] = cidade?.toJson();
    _data['usuario_administrador'] = usuarioAdministrador?.toJson();
    _data['nome_usuario'] = nomeUsuario;
    _data['horario_abertura'] = horarioAbertura;
    _data['horario_fechamento'] = horarioFechamento;
    return _data;
  }
}

File: backEnd/Systetica_Api/src/main/java/com/frfs/systetica/entity/Empresa.java
package com.frfs.systetica.entity;

import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "empresa")
public class Empresa implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(name = "nome", length = 150)
    private String nome;

    @NotNull
    @Column(name = "cnpj", length = 18, unique = true)
    private String cnpj;

    @NotNull
    @Column(name = "telefone1", length = 15)
    private String telefone1;

    @Column(name = "telefone2", length = 15)
    private String telefone2;

    @NotNull
    @Column(name = "endereco", length = 100)
    private String endereco;

    @NotNull
    @Column(name = "numero", length = 8)
    private int numero;

    @NotNull
    @Column(name = "cep", length = 9)
    private String cep;

    @NotNull
    @Column(name = "bairro", length = 100)
    private String bairro;

    @Column(name = "latitude", length = 13)
    private String latitude;

    @Column(name = "longitude", length = 13)
    private String longitude;

    @Column(name = "logo_base64")
    @Lob
    private String logoBase64;

    @NotNull
    @Column(name = "data_cadastro")
    private Date dataCadastro;

    @NotNull
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_cidade")
    private Cidade cidade;

    @NotNull
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_usuario_administrador")
    private Usuario usuarioAdministrador;

    @ManyToMany()
    @JoinTable(
            name = "ass_funcionario_empresa",
            uniqueConstraints = @UniqueConstraint(columnNames = {"id_empresa", "id_usuario"}),
            joinColumns = @JoinColumn(name = "id_empresa"),
            inverseJoinColumns = @JoinColumn(name = "id_usuario"))
    private List<Usuario> usuarios = new ArrayList<>();
}